src/printer.js

```diff
@@ -1828,15 +1832,33 @@ function printExportDeclaration(path, options, print) {
     } else {
       parts.push(
         decl.exportKind === "type" ? "type " : "",
-        shouldPrintSpaces ? "{ " : "{",
-        join(", ", path.map(print, "specifiers")),
-        shouldPrintSpaces ? " }" : "}"
+        conditionalGroup([
```

Comment:
You don't need `conditionalGroup` here. You only need that in very advanced cases. Not that it's only used in 3 or 4 other places.

This one can be broken like an object with a normal group

```js
group(concat([
  "{",
  indent(
    options.tabWidth,
    concat([
      hardline,
      join(concat([",", options.bracketSpacing ? line : softline]),
           path.map(print, "specifiers"))
    ])
  ),
  options.bracketSpacing ? line : softline,
  "}"
]))
```

That code may not by fully correct, but it's mostly there.